src/app/scripts/cac/control/cac-control-directions-list.js

```diff
@@ -207,10 +230,122 @@ CAC.Control.DirectionsList = (function ($, Handlebars, UserPreferences, Utils) {
         }
     }
 
+    function shareOnFacebook(event) {
+        var caption = 'Trip Plan on GoPhillyGo';
+
+        // TODO: get a screenshot of the map page to post?
+        var pictureUrl = [options.useHost,
+                          '/static/images/logo_color.svg'
+                         ].join('');
+
+        if (typeof FB !== 'undefined') {
+            // prompt user to log in, if they aren't already
+            FB.getLoginStatus(function(response) {
+                if (response.status !== 'connected') {
+                    FB.login();
+                }
+            });
+
+            FB.ui({
+                method: 'feed',
+                link: event.data.url,
+                caption: caption,
+                picture: pictureUrl,
+            }, function(response){
+                if (!response || _.has(response, 'error_code')) {
+                    console.warn(response);
+                    console.warn('did not post to facebook');
+                }
+            });
+        } else {
+            console.warn('FB unavailable. Is script loaded?');
+            // TODO: redirect to URL if API unavailable
+
+            var feedUrl = 'https://www.facebook.com/dialog/feed?';
+
+            /* jshint camelcase:false */
+            var params = {
+                app_id: Settings.fbAppId,
+                display: 'popup',
+                caption: caption,
+                picture: pictureUrl,
+                link: event.data.url,
+                redirect_uri: options.useHost + '/map'
+            };
+            /* jshint camelcase:true */
+
+            var url = feedUrl + $.param(params);
+            window.open(url, '_blank');
+            event.returnValue = false;
```

Comment:
You mean setting the `event.returnValue` and `preventDefault`? Nope, not necessary, but doesn't hurt either.  If the button somehow got changed to a submit type, it would be needed.  Those calls came from the documentation boilerplate from both Facebook and Twitter, so I assume they serve some purpose in some cases.